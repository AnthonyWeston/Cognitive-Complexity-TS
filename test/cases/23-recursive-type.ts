type A = {
    attr: A;
};

type B = {
    attr: A;
};

interface C {
    attr: C;
};

interface D {
    attr: C;
};

class E {
    attr: E;
}