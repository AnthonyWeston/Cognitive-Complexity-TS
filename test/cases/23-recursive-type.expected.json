{
    "23-recursive-type.ts": {
        "score": 3,
        "inner": [
            {
                "name": "A",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "B",
                "score": 0,
                "line": 5,
                "column": 1
            },
            {
                "name": "C",
                "score": 1,
                "line": 9,
                "column": 1
            },
            {
                "name": "D",
                "score": 0,
                "line": 13,
                "column": 1
            },
            {
                "name": "E",
                "score": 1,
                "line": 17,
                "column": 1
            }
        ]
    }
}